src/types.py

```diff
@@ -239,67 +311,139 @@ def get_function_pointer_signature(self) -> Optional["FunctionSignature"]:
                 return ptr_to.fn_sig
         return None
 
-    def get_ctype_fields(
+    def get_struct_declaration(self) -> Optional["StructDeclaration"]:
+        """If self is a struct, return the StructDeclaration"""
+        if self.is_struct():
+            data = self.data()
+            assert data.struct is not None
+            return data.struct
+        return None
+
+    _GetFieldResult = Tuple[AccessPath, "Type", Optional[int]]
+
+    def get_field(
```

Comment:
Some high-level comments about the new `get_field`:
- Replaces both the old `get_field()` & `find_substruct_array()`
- Recursively traverses struct fields/arrays, rather than using a precomputed set of fields.
- The return type is kind of awkward, still considering how best to refactor that.